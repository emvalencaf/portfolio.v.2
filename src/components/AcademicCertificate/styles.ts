import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

type WrapperAcademicCertificateProps = {
	theme: DefaultTheme;
	showDetails: boolean;
}

const changeCursor = () => css`
	cursor: pointer;
`;

export const Wrapper = styled.div<WrapperAcademicCertificateProps>`
	${({ theme, showDetails }) => css`
	padding: 2rem;
	${!showDetails && changeCursor()}

	& ${HeadingContainer} {
		margin: 0px;
	}
	`}
`;

export const HeaderAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		margin: ${theme.spacings.small} auto;
	`}
`;

export const BodyAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.small};
		margin: ${theme.spacings.small} auto;
	`}
`;