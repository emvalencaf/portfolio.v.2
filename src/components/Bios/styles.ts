import styled, { css } from "styled-components";
export const Wrapper = styled.div`
	${({ theme }) => css`
		@media ${theme.media.lteMedium} {
			flex-direction: column;
		}

		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: space-between;
		align-items: center;
		gap: ${theme.spacings.xhero};
		text-align: justify;
		& p {
			text-align: justify;
			padding: ${theme.spacings.small};
			max-width: 550px;
		}
	`}
`;

export const ContentContainer = styled.div`
	${() => css`
		max-height: 350px;
		overflow-y: auto;
	`}
`;
