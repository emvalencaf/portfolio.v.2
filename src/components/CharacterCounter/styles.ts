import styled, { css } from "styled-components";

export const Wrapper = styled.div`
	${() => css`
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-items: end;
	`}
`;
