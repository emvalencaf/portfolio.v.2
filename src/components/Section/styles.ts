import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
	${() => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		height: 500px;
		max-height: 500px;
		padding: 0px 2.5rem;
	`}
`;
