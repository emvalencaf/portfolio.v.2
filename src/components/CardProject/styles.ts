import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

export const Wrapper = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: space-between;
		height: ${theme.frameSizes.medium};
		width: ${theme.frameSizes.smallMedium};
		border-radius: 2.9rem;
		padding: ${theme.spacings.small};
		background-color: ${theme.colors.gray1};
		transition: ${theme.transitions.normal};
		opacity: 0.4;
		&:focus,
		&:hover {
			opacity: 1;
			box-shadow: 0 0 ${theme.spacings.medium} ${theme.spacings.tiny} ${theme.colors.gray6};
			filter: brightness(110%);
		}
	`}
`;

export const PictureContainer = styled.picture`
	${({ theme }) => css`
		@keyframes photo {
			0% {
				transform: scale(0);
			}
			50% {
				transform: scale(1.1);
			}
			100% {
				transform: scale(1);
			}
		}

		display: flex;
		align-items: center;
		justify-content: center;
		width: 100%;
		transform: scale(0);
		transition: all ${theme.transitions.normal} ease-in-out;
		animation: photo ${theme.transitions.slowest};
		animation-fill-mode: forwards;

		> img {
			width: 100%;
			height: 100%;
			margin: auto;
			border-radius: 2rem;
		}
	`}
`;

export const ContentContainer = styled.p`
	${({ theme }) => css`
		@keyframes content {
			0% {
				transform: scale(0);
			}
			50% {
				transform: scale(1.1);
			}
			100% {
				transform: scale(1);
			}
		}
		transform: scale(0);
		transition: all ${theme.transitions.normal} ease-in-out;
		animation: content ${theme.transitions.slowest};
		animation-fill-mode: forwards;
	`}
`;

export const HeaderCardContainer = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-content: space-between;
		width: 100%;
		background-color: ${theme.colors.gray4};
		border-radius: 2rem;
		opacity: 0.7;
		z-index: ${theme.layers.layer3};

		& ${HeadingContainer} {
			text-align: center;
			font-size: ${theme.fonts.sizes.normal};
			color: ${theme.colors.gray6};
			margin: 0px;
			padding: ${theme.spacings.xxsmall};
		}

		& svg {
			height: ${theme.fonts.sizes.medium};
			width: ${theme.fonts.sizes.medium};
			transition: all ${theme.transitions.normal} ease-in-out;
			margin: ${theme.spacings.small};
			&:hover {
				color: ${theme.colors.primary};
			}
		}

		> span > svg {
			@keyframes rotate {
				0% {
					transform: rotate(90deg);
				}
				100% {
					transform: rotate(180deg);
				}
				100% {
					transform: rotate(360deg);
				}
			}
			cursor: pointer;
			&:hover {
				animation: rotate ${theme.transitions.slowest} infinite;
			}
		}
	`}
`;

export const ButtonsContainer = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: row;
		gap: ${theme.spacings.small};
		> a {
			background-color: ${theme.colors.gray6};
			color: ${theme.colors.white};
			font-size: ${theme.fonts.sizes.normal};
			padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
			cursor: pointer;
			border-radius: ${theme.spacings.tiny};
			transition: all ${theme.transitions.fast} ease-in-out;
			display: flex;
			align-items: center;
			justify-content: center;
			text-decoration: none;

			> svg {
				width: 2rem;
				height: 2rem;
				margin-left: 1rem;
				margin-right:1rem;
			}

			&:focus {
				outline: none;
				box-shadow: 0 0 ${theme.spacings.tiny} ${theme.colors.primary};
				filter: brightness(110%);
			}

			&:hover {
				filter: brightness(90%);
			}
		}
	`}
`;