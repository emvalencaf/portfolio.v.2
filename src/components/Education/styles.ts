import styled, { css } from 'styled-components';
import { Button } from '../Button/styles';

export const Wrapper = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		gap: ${theme.spacings.mediumSmall};
	`}
`;

export const OrderList = styled.ol`
	${({ theme }) => css`
		margin: 0;
		padding: 0;
		list-style: none;
		max-width: 400px;

		> li {
			padding: 0px ${theme.spacings.mediumSmall};
		}

		> li:has(> button:focus) {
			list-style: disc;
			color: ${theme.colors.primary};
		}

		& ${Button} {
			align-items: flex-start;
			justify-content: flex-start;
			color: ${theme.colors.primary};
			background-color: transparent;
			font-size: ${theme.fonts.sizes.small};
			text-align: justify;
			border: none;
			outline: none;
			width: 100%;
		}

		& ${Button}:active,
		& ${Button}:focus,
		& ${Button}:hover {
			box-shadow: none;
		}

	`}
`;

export const Content = styled.div`

`;