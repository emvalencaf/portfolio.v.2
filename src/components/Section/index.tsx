// styles
import * as Styled from './styles';

// types
export type SectionProps = {
	children: React.ReactNode;
};

const Section = ({ children }: SectionProps) => {
	return (
		<Styled.Wrapper>
			{ children }
		</Styled.Wrapper>
	);
};

export default Section;
