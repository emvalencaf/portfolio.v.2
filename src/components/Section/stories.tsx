import { Meta, Story } from "@storybook/react/types-6-0";
import Section, { SectionProps } from ".";

export default {
	title: "Section",
	component: Section,
	args: {
		children: "<h1> Ol√° </h1>",
	},
} as Meta<SectionProps>;

export const Template: Story<SectionProps> = (args) => {
	return (
		<div>
			<Section {...args} />
		</div>
	);
};
