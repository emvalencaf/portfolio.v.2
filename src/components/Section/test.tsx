import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";

// react component to be test
import Section from ".";

describe("<Section />", () => {
	it("should render component it props default values", () => {
		renderTheme(
			<Section id="#">
				<h1>Olá</h1>
			</Section>
		);

		expect(
			screen.getByRole("heading", { name: "Olá" })
		).toBeInTheDocument();
	});

	it("should match snapshot", () => {
		const { container } = renderTheme(
			<Section id="#">
				<h1>Olá</h1>
			</Section>
		);

		expect(container).toMatchSnapshot();
	});
});
