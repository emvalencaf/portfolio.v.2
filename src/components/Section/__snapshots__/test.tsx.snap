// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 500px;
  max-height: 500px;
  padding: 0px 2.5rem;
}

<div>
  <div
    class="c0"
  >
    <h1>
      Ol√°
    </h1>
  </div>
</div>
`;
